server/athenian/api/align/goals/dbaccess.py

```diff
@@ -71,3 +124,49 @@ async def _validate_goal_creation_info(
         raise GoalMutationError(
             f"Some teamId-s don't exist or access denied: {missing_repr}", HTTPStatus.FORBIDDEN,
         )
+
+
+async def _validate_team_goal_deletions(
+    account_id: int, goal_id: int, team_ids: Sequence[int], sdb_conn: DatabaseLike,
+) -> None:
+    where_clause = sa.and_(
+        Goal.account_id == account_id,
+        Team.owner_id == account_id,
+        TeamGoal.goal_id == goal_id,
+        TeamGoal.team_id.in_(team_ids),
+    )
+    select_stmt = sa.select(
+        [TeamGoal.team_id],
+    ).join_from(TeamGoal, Goal).join(Team).where(where_clause)
```

Comment:
It's not obvious by which fields we join + we specify the same account_id condition twice. This code should perform better:
``` python
sa.select([TeamGoal.team_id]).select_from(join(join(TeamGoal, Goal, TeamGoal.goal_id == Goal.id)), Team, sa.and_(Goal.account_id == Team.owner_id, TeamGoal.team_id == Team.id)).where(sa.and_(Goal.account_id == account_id, TeamGoal.goal_id == goal_id, TeamGoal.team_id.in_(team_ids)))
```
Maybe we need a few indexes added for `account_id`/`owner_id`. We need them in any case.

Reply:
In your version the condition is moved from `WHERE` to `JOIN` `ON`, but the plan it's the same being an inner join:
```

Reply:
SQLAlchemy can detect join condition automatically given fk declarations but if you prefer I can write explicit conditions.